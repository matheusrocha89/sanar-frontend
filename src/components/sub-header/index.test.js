import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import SubHeader from './';


describe('SubHeader', () => {
  it('renders correctly', () => {
    const addNewExam = jest.fn();
    const tree = renderer.create(<SubHeader addNewExam={addNewExam} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Hide modal', () => {
    const addNewExam = jest.fn();
    const comp = shallow(<SubHeader addNewExam={addNewExam} />);
    comp.instance().handleClose();
    expect(comp.state().showModal).toBeFalsy();
  });

  it('Shows modal', () => {
    const addNewExam = jest.fn();
    const comp = shallow(<SubHeader addNewExam={addNewExam} />);
    comp.instance().handleShow();
    expect(comp.state().showModal).toBeTruthy();
  });
});
