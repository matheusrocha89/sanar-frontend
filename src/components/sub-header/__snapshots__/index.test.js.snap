// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader renders correctly 1`] = `
<div
  className="SubHeader clearfix"
>
  <div
    className="SubHeader-title-wrapper"
  >
    <h4>
      Provas
    </h4>
  </div>
  <div
    className="SubHeader-add-button-wrapper"
  >
    <button
      className="btn btn-success"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Novo
    </button>
  </div>
</div>
`;
